[2025-05-26T18:25:30.546+0300] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2025-05-26T18:25:30.561+0300] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: currency_etl_dag.fetch_and_store_currency manual__2025-05-26T15:20:16.216480+00:00 [queued]>
[2025-05-26T18:25:30.566+0300] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: currency_etl_dag.fetch_and_store_currency manual__2025-05-26T15:20:16.216480+00:00 [queued]>
[2025-05-26T18:25:30.566+0300] {taskinstance.py:2306} INFO - Starting attempt 2 of 2
[2025-05-26T18:25:30.585+0300] {taskinstance.py:2330} INFO - Executing <Task(PythonOperator): fetch_and_store_currency> on 2025-05-26 15:20:16.216480+00:00
[2025-05-26T18:25:30.588+0300] {standard_task_runner.py:63} INFO - Started process 18920 to run task
[2025-05-26T18:25:30.591+0300] {standard_task_runner.py:90} INFO - Running: ['airflow', 'tasks', 'run', 'currency_etl_dag', 'fetch_and_store_currency', 'manual__2025-05-26T15:20:16.216480+00:00', '--job-id', '7', '--raw', '--subdir', 'DAGS_FOLDER/currency_dag.py', '--cfg-path', '/tmp/tmpgzssydro']
[2025-05-26T18:25:30.593+0300] {standard_task_runner.py:91} INFO - Job 7: Subtask fetch_and_store_currency
[2025-05-26T18:25:30.641+0300] {task_command.py:426} INFO - Running <TaskInstance: currency_etl_dag.fetch_and_store_currency manual__2025-05-26T15:20:16.216480+00:00 [running]> on host Home.
[2025-05-26T18:25:30.698+0300] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='airflow' AIRFLOW_CTX_DAG_ID='currency_etl_dag' AIRFLOW_CTX_TASK_ID='fetch_and_store_currency' AIRFLOW_CTX_EXECUTION_DATE='2025-05-26T15:20:16.216480+00:00' AIRFLOW_CTX_TRY_NUMBER='2' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-05-26T15:20:16.216480+00:00'
[2025-05-26T18:25:30.700+0300] {taskinstance.py:430} INFO - ::endgroup::
[2025-05-26T18:25:31.461+0300] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2025-05-26T18:25:31.461+0300] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/imsim/currency_pipeline/venv/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/imsim/currency_pipeline/venv/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/imsim/currency_pipeline/venv/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 400, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/imsim/currency_pipeline/venv/lib/python3.12/site-packages/airflow/operators/python.py", line 235, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/imsim/currency_pipeline/venv/lib/python3.12/site-packages/airflow/operators/python.py", line 252, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/imsim/currency_pipeline/airflow/dags/fetch_currency_data.py", line 18, in fetch_and_store_currency
    cur.execute("""
psycopg2.errors.InsufficientPrivilege: permission denied for schema public
LINE 2:         CREATE TABLE IF NOT EXISTS currency_rates (
                                           ^

[2025-05-26T18:25:31.468+0300] {taskinstance.py:1206} INFO - Marking task as FAILED. dag_id=currency_etl_dag, task_id=fetch_and_store_currency, run_id=manual__2025-05-26T15:20:16.216480+00:00, execution_date=20250526T152016, start_date=20250526T152530, end_date=20250526T152531
[2025-05-26T18:25:31.487+0300] {standard_task_runner.py:110} ERROR - Failed to execute job 7 for task fetch_and_store_currency (permission denied for schema public
LINE 2:         CREATE TABLE IF NOT EXISTS currency_rates (
                                           ^
; 18920)
[2025-05-26T18:25:31.528+0300] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2025-05-26T18:25:31.539+0300] {taskinstance.py:3498} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-05-26T18:25:31.539+0300] {local_task_job_runner.py:222} INFO - ::endgroup::
